src/native/fs/fs_napi.cpp

```diff
@@ -219,6 +230,258 @@ struct Stat : public FSWorker
     }
 };
 
+/**
+ * Getxattr is an fs op
+ */
+struct Getxattr : public FSWorker
+{
+    std::string _path;
+    std::vector<KeyValue> _md;
+    Getxattr(const Napi::CallbackInfo& info) 
+        : FSWorker(info)
+        , _md(0)
+    {
+        _path = info[1].As<Napi::String>();
+        Begin(XSTR() << "Getxattr " << DVAL(_path));
+    }
+    virtual void Work()
+    {
+        ssize_t buf_len, key_len, value_len;
+        char *buf, *key, *val;
+
+        #if !defined(_POSIX_C_SOURCE) || defined(_DARWIN_C_SOURCE)
+                buf_len = listxattr(_path.c_str(), NULL, 0, 0);
+        #else
+                buf_len = listxattr(_path.c_str(), NULL, 0);
+        #endif
+
+        if (buf_len <= 0) {
+            if (buf_len == -1) {
+                SetSyscallError();
+            }
+            return;
+        }
+
+        struct stat stat_res;
+        auto r = stat(_path.c_str(), &stat_res);
+        if (r) {
+            SetSyscallError();
+            return;
+        }
+        auto start_ctime = stat_res.st_ctime;
+
+        buf = static_cast<char*>(malloc(buf_len));
```

Comment:
This is leak prone.
For safe memory allocation use `Buf buf(len)` - see [here](https://github.com/noobaa/noobaa-core/blob/master/src/native/util/buf.h)
For more complex cases you can use [StackCleaner](https://github.com/noobaa/noobaa-core/blob/f1b3a1b40262b88a01a647483d0d097d8813f777/src/native/util/common.h#L120), but this case is simple